rego/rego.go

```diff
@@ -545,24 +626,153 @@ func (r *Rego) Compile(ctx context.Context) (*CompileResult, error) {
 	return result, nil
 }
 
-func (r *Rego) parse() (map[string]*ast.Module, ast.Body, error) {
+// PrepareFlag defines an enum type to control the behavior
+// of the Prepare() call.
+type PrepareFlag int
+
+// PrepareFlags for use in Prepare()
+const (
+	WithPartialEval PrepareFlag = iota
+)
+
+// Prepare will parse inputs, modules, and query arguments in preparation
+// of evaluating them.
+// If `partialEval` is true this will partially evaluate
+func (r *Rego) Prepare(ctx context.Context, flags ...PrepareFlag) (PreparedQuery, error) {
```

Comment:
With the enum/flag interface, how would parameterize the Prepare call with more information? E.g., what if we wanted to provide an optional set of string parameters with the partial eval flag?

Reply:
We couldn't.. 😟  Is that a thing we want to do?

I was working on the assumption that parameters would be passed into rego.New() and selective ones later on in the PreparedQuery function. I'm open to making this more flexible though if you want. Technically there isn't a reason it can't take in the same options that rego.New() aside from I didn't plumb it through.